tron/core/jobrun.py

```diff
@@ -228,25 +231,25 @@ def handle_action_run_state_change(self, action_run, _):
         if action_run.is_skipped and self.action_runs.is_scheduled:
             return
 
-        if not action_run.is_broken and any(self._start_action_runs()):
-            log.info("Action runs started for %s." % self)
-            return
+        if not action_run.is_broken:
+            started = self._start_action_runs()
+            if any(started):
+                log.info(
+                    f"{self} action runs started: "
+                    f"{', '.join(str(s) for s in started)}"
+                )
+                return
 
         if self.action_runs.is_active or self.action_runs.is_scheduled:
-            log.info("%s still has running or scheduled actions." % self)
+            log.info(f"{self} still has running or scheduled actions")
             return
 
         # If we can't make any progress, we're done
-        cleanup_run = self.action_runs.cleanup_action_run
+        cleanup_run: ActionRun = self.action_runs.cleanup_action_run
         if not cleanup_run or cleanup_run.is_done:
             return self.finalize()
 
-        # TODO: remove in (0.6), start() no longer raises an exception
-        # When a job is being disabled, or the daemon is being shut down a bunch
-        # of ActionRuns will be cancelled/failed. This would cause cleanup
-        # action to be triggered more then once. Guard against that.
-        if cleanup_run.check_state('start'):
-            cleanup_run.start()
+        cleanup_run.start()
```

Comment:
No need to do clenup_run.check_state('start') anymore?

Reply:
that's what i  understood from the removed comment just above the `if`:

     # TODO: remove in (0.6), start() no longer raises an exception
    # When a job is being disabled, or the daemon is being shut down a bunch
    # of ActionRuns will be cancelled/failed. This would cause cleanup
    # action to be triggered more then once. Guard against that.

Reply:
Ah, gotcha.